### PR TITLE
feat: add task header downloader

### PR DESCRIPTION
**Pull request by mattsse**
Introduces a new `HeaderDownloader` that spawns a downloader to a task.

This way the actual downloader can operate independently from the Stage and isn't affected by DB commit workload.

blocked by #957

### CODE REVIEW

**Comment by onbjerg** [comment]
Note: Need this for bodies too (if not already the case in #905)

**Comment by mattsse** [comment]
Needs followup to integrate in pipeline Builder

File: crates/net/downloaders/src/headers/task.rs [hunk 1]
```diff
@@ -0,0 +1,200 @@
+//!
```
Comment (rkrasiuk) [inline crates/net/downloaders/src/headers/task.rs:1]: nit: add docs or remove

File: crates/net/downloaders/src/headers/test_utils.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+use once_cell::sync::Lazy;
+use reth_interfaces::{consensus::Consensus, test_utils::TestConsensus};
+use reth_primitives::SealedHeader;
+use std::sync::Arc;
+
+pub(crate) static CONSENSUS: Lazy<Arc<dyn Consensus>> =
+    Lazy::new(|| Arc::new(TestConsensus::default()));
```
Comment (rkrasiuk) [inline crates/net/downloaders/src/headers/test_utils.rs:7]: let's remove this altogether

**Review by rkrasiuk — APPROVED** [review]
lgtm, left small comments

**Pull request merged**